File: big_O/javascript/1_find_nemo.js
const name = ['nemo'];
const names = ['cristofer', 'oliver', 'josue', 'monse', 'gerardo', 'pollo', 
                'gallo'];
const large = new Array(100000).fill('pollo')

function findNemo(array) {
    let t0 = performance.now(); // O(1)
    for (let i = 0; i <= array.length - 1; i++) { // O(n)
        if (array[i] === 'nemo') { // O(n)
            console.log('We found Nemo!'); // O(n)
        }
    }
    let t1 = performance.now(); // O(1)
    console.log(`${t1 - t0} milliseconds`); // O(1)
}

findNemo(large); // O(n)

File: big_O/python/1_find_nemo.py
import timeit

code_to_test = """
large = ["pollo" for value in range(100)]
for value in large:
    if value == 'nemo':
        print('We found Nemo!')
"""

elapsed_time = timeit.timeit(code_to_test, number=1) / 100 
print(elapsed_time)